pkg/utils/utils.go

```diff
@@ -116,15 +116,20 @@ func ToDNS1123Subdomain(name string) string {
 // The name's length will be short enough to be valid for K8s Services.
 func GenerateFixedName(owner metav1.Object, prefix string) string {
```

Comment:
This method should be deleted and use https://github.com/knative/pkg/blob/master/kmeta/names.go#L39

Reply:
thanks @n3wscott 

`GenerateFixedName` is used by 

- EventType
- APIServerSource ReceiveAdapter
- Subscriptions 
- PingSource ReceiveAdapter

 I'll broaden the scope of this PR to exceed the initial Trigger-Subscription issue and replace all uses of `GenerateFixedName`

Reply:
We need to be careful when changing how names are generated. Will kmeta.ChildName always generate the same result as GenerateFixedName?

Reply:
Yes, the point of `kmeta.ChildName` is to do just this and be dns compliant. it is used in serving to make revisions. @vagababov

Reply:
I think is slightly different. So you need to be certain it's backward compatible, but our names are guaranteed valid k8s object names (which implies DNS valid name).

Reply:
yes, it is different. This is the result of the `GenerateFixedName` test cases adapted to the `TestChildName` tests:

```
--- FAIL: TestChildName (0.00s)
    --- FAIL: TestChildName/default-text-extractor-2d6c09e1-aa54-11e9-9d6a-42010a8a0062 (0.00s)
        names_test.go:94: default-text-extractor-2d6c09e1-aa54-11e9-9d6a-42010a8a0062: got:      default-text-extractor2d6c09e1-aa54-11e9-9d6a-42010a8a0062 want:     default-text-extractor-2d6c09e1-aa54-11e9-9d6a-42010a8a0062
            diff:  string(
            -   "default-text-extractor-2d6c09e1-aa54-11e9-9d6a-42010a8a0062",
            +   "default-text-extractor2d6c09e1-aa54-11e9-9d6a-42010a8a0062",
              )
    --- FAIL: TestChildName/this-is-an-extremely-long-prefix-which-will-make-the-generated-name-too-long--2d6c09e1-aa54-11e9-9d6a-42010a8a0062 (0.00s)
        names_test.go:94: this-is-an-extremely-long-prefix-which-will-make-the-generated-name-too-long--2d6c09e1-aa54-11e9-9d6a-42010a8a0062: got: this-is-an-extremely-long-prefi65d0b085fde126ba79fa7952fcbf7345 want: this-is-an-extremely-long--2d6c09e1-aa54-11e9-9d6a-42010a8a0062
            diff:  string(
            -   "this-is-an-extremely-long--2d6c09e1-aa54-11e9-9d6a-42010a8a0062",
            +   "this-is-an-extremely-long-prefi65d0b085fde126ba79fa7952fcbf7345",
              )
    --- FAIL: TestChildName/this-is-an-extremely-long-prefix-which-will-make-the-generated-name-too-long---2d6c09e1-aa54-11e9-9d6a-42010a8a0062 (0.00s)
        names_test.go:94: this-is-an-extremely-long-prefix-which-will-make-the-generated-name-too-long---2d6c09e1-aa54-11e9-9d6a-42010a8a0062: got: this-is-an-extremely-long-prefi18f0261bb667183f28b9caabd7a9ab81 want: this-is-an-extremely-long--2d6c09e1-aa54-11e9-9d6a-42010a8a0062
            diff:  string(
            -   "this-is-an-extremely-long--2d6c09e1-aa54-11e9-9d6a-42010a8a0062",
            +   "this-is-an-extremely-long-prefi18f0261bb667183f28b9caabd7a9ab81",
              )
    --- FAIL: TestChildName/default-text-extractor.-2d6c09e1-aa54-11e9-9d6a-42010a8a0062 (0.00s)
        names_test.go:94: default-text-extractor.-2d6c09e1-aa54-11e9-9d6a-42010a8a0062: got:     default-text-extractor.2d6c09e1-aa54-11e9-9d6a-42010a8a0062 want:     default-text-extractor-2d6c09e1-aa54-11e9-9d6a-42010a8a0062
            diff:  string(
            -   "default-text-extractor-2d6c09e1-aa54-11e9-9d6a-42010a8a0062",
            +   "default-text-extractor.2d6c09e1-aa54-11e9-9d6a-42010a8a0062",
              )
    --- FAIL: TestChildName/this-is-an-extremely-long.prefix-and-it-will-be-cut-at-the-dot-2d6c09e1-aa54-11e9-9d6a-42010a8a0062 (0.00s)
        names_test.go:94: this-is-an-extremely-long.prefix-and-it-will-be-cut-at-the-dot-2d6c09e1-aa54-11e9-9d6a-42010a8a0062: got: this-is-an-extremely-long.prefi1ef5a760f3cded7e0761416d666e5cde want:  this-is-an-extremely-long-2d6c09e1-aa54-11e9-9d6a-42010a8a0062
            diff:  string(
            -   "this-is-an-extremely-long-2d6c09e1-aa54-11e9-9d6a-42010a8a0062",
            +   "this-is-an-extremely-long.prefi1ef5a760f3cded7e0761416d666e5cde",
              )
```

either we

1. keep current `GenerateFixedName` with the fix at this PR
2. use `ChildName` making next release incompatible
3. Something ugly, like adding for the 4 cases above a migration helper that will look for new and old expected resource names when reconciling, if they don't exist we create new ones using `kmeta.ChildName`. Release an incompatibility note to remove the hack in future versions.

Reply:
We did twofer in serving for 2 releases and then removed the old code.
Like you get new name, if it's missing old name. Not sure if it'll work here :)

Reply:
Sure Victor It might, that would be option 3, ugly but a smooth transition.
WDYT @n3wscott @grantr ?

Reply:
For all cases it looks for an object by the generated name, and if it doesn't exists it will create it.
Users would end up with duplicated deployments (receive adapters) and subscriptions.

We don't need a migration strategy merging this PR, it is backwards compatible but for cases where current implementation generates invalid DNS names and fails.

My take on this would be merging this, it is fixing current issue, no matter if we get rid of `GenerateFixedName`. Then open a follow up issue, to discuss how to remove `GeneratedFixedNames`.

Reply:
The difference between these cases and serving though is that the listed cases:

```
EventType
APIServerSource ReceiveAdapter
Subscriptions
PingSource ReceiveAdapter
```

Do not use the generated name to do name resolution. If you do 2. it seems like it will be fine operationally with a big gotchya: we use that name to query the api server for the resource the controller is attempting to make and if we find it and don't own it, then it is an error. If we find it, we update. If we don't find it we create it. The last point will cause duplicates of all sub-resources.

How do you like 3. with a twist: query with `ChildName` first, if not found, query for `GenerateFixedName`. If found with `GenerateFixedName` then delete it and treat is as the same case as not found from `ChildName`.
 This will cause a moment of downtime and the subscriptions will drop state if they have it...

Reply:
That's the pattern I was thinking for case 3, there is no way of changing object meta without re-creating so that's it.

At first there doesn't seem to be an easy way of abstracting that double checks for `GeneratedFixedName` and `ChildName` existence but I'll try to come up with the a solution that doesn't spread much code around. I'll work on a different PR and park/remove this one if that's ok with you.

Reply:
> My take on this would be merging this, it is fixing current issue, no matter if we get rid of GenerateFixedName. Then open a follow up issue, to discuss how to remove GeneratedFixedNames.

I agree, if there's no backward incompatible change here, then deprecating `GenerateFixedName` is out of scope for this PR. I created #2842 to track the deprecation. Please continue the conversation there :)